janitor/finance.py

```diff
@@ -204,3 +487,291 @@ def convert_currency(
+        'Nicaragua': 'NIC', 'Netherlands': 'NLD', 'Norway': 'NOR',
+        'Nepal': 'NPL', 'Nauru': 'NRU', 'New Zealand': 'NZL',
+        'OECD members': 'OED', 'Oman': 'OMN',
+        'Other small states': 'OSS', 'Pakistan': 'PAK',
+        'Panama': 'PAN', 'Peru': 'PER', 'Philippines': 'PHL',
+        'Palau': 'PLW', 'Papua New Guinea': 'PNG', 'Poland': 'POL',
+        'Pre-demographic dividend': 'PRE', 'Puerto Rico': 'PRI',
+        "Korea, Dem. People's Rep.": 'PRK', 'Portugal': 'PRT',
+        'Paraguay': 'PRY', 'West Bank and Gaza': 'PSE',
+        'Pacific island small states': 'PSS',
+        'Post-demographic dividend': 'PST',
+        'French Polynesia': 'PYF', 'Qatar': 'QAT',
+        'Romania': 'ROU', 'Russian Federation': 'RUS',
+        'Rwanda': 'RWA', 'South Asia': 'SAS',
+        'Saudi Arabia': 'SAU', 'Sudan': 'SDN', 'Senegal': 'SEN',
+        'Singapore': 'SGP', 'Solomon Islands': 'SLB',
+        'Sierra Leone': 'SLE', 'El Salvador': 'SLV',
+        'San Marino': 'SMR', 'Somalia': 'SOM', 'Serbia': 'SRB',
+        'Sub-Saharan Africa (excluding high income)': 'SSA',
+        'South Sudan': 'SSD', 'Sub-Saharan Africa': 'SSF',
+        'Small states': 'SST', 'Sao Tome and Principe': 'STP',
+        'Suriname': 'SUR', 'Slovak Republic': 'SVK',
+        'Slovenia': 'SVN', 'Sweden': 'SWE', 'Eswatini': 'SWZ',
+        'Sint Maarten (Dutch part)': 'SXM', 'Seychelles': 'SYC',
+        'Syrian Arab Republic': 'SYR',
+        'Turks and Caicos Islands': 'TCA', 'Chad': 'TCD',
+        'East Asia & Pacific (IDA & IBRD countries)': 'TEA',
+        'Europe & Central Asia (IDA & IBRD countries)': 'TEC',
+        'Togo': 'TGO', 'Thailand': 'THA', 'Tajikistan': 'TJK',
+        'Turkmenistan': 'TKM',
+        'Latin America & the Caribbean (IDA & IBRD countries)': 'TLA',
+        'Timor-Leste': 'TLS',
+        'Middle East & North Africa (IDA & IBRD countries)': 'TMN',
+        'Tonga': 'TON', 'South Asia (IDA & IBRD)': 'TSA',
+        'Sub-Saharan Africa (IDA & IBRD countries)': 'TSS',
+        'Trinidad and Tobago': 'TTO', 'Tunisia': 'TUN',
+        'Turkey': 'TUR', 'Tuvalu': 'TUV', 'Tanzania': 'TZA',
+        'Uganda': 'UGA', 'Ukraine': 'UKR',
+        'Upper middle income': 'UMC',
+        'Uruguay': 'URY', 'United States': 'USA',
+        'Uzbekistan': 'UZB', 'St. Vincent and the Grenadines': 'VCT',
+        'Venezuela, RB': 'VEN', 'British Virgin Islands': 'VGB',
+        'Virgin Islands (U.S.)': 'VIR', 'Vietnam': 'VNM',
+        'Vanuatu': 'VUT', 'World': 'WLD', 'Samoa': 'WSM',
+        'Kosovo': 'XKX', 'Yemen, Rep.': 'YEM', 'South Africa': 'ZAF',
+        'Zambia': 'ZMB', 'Zimbabwe': 'ZWE'}
+    :param currency_year: The currency year to inflate from.
+        The year should be 1960 or later.
+    :param to_year: The currency year to inflate to.
+        The year should be 1960 or later.
+    :param make_new_column: Generates new column for inflated currency if
+        True, otherwise, inflates currency in place.
+
+    :Setup:
```

Comment:
The structure "setup-example-output" is, at the moment, done in Jupyter notebooks. In the API docs, in which the docstrings are rendered, we only show an example of how to call the function, without showing the necessary inputs and outputs. Would you be kind enough to move this example to a notebook instead, so that the docstrings can be standardized on just showing the method-chaining and functional examples?

Reply:
@ericmjl Just to be clear, are you suggesting that:

1. I pull out the setup and output portions of the doc string entirely (leave in the example), and
2. Put all three (setup, example, and output) in a separate notebook located under /examples/notebooks.

Reply:
@rahosbach yes, that’s in the right direction. Maybe a more specific thing would be:

- Remove setup and output portions.
- Follow the rest of the examples in the biology and chemistry submodules, where I know that @loganthomas has started standardizing the docstrings (see PR #472 and #461). That is where we would like to be:
    1. One line description
    1. Additional notes
    1. Method chaining usage example
    1. Functional usage example
    1. Parameter description
    1. Return description

If you would like to have an example available, my loosely held opinion is that the best place is the notebook examples. My other loosely held opinion is that docstrings should be as small as possible, but not smaller than the aforementioned list. Hence, real-life examples go in notebooks, rather than the docstrings. This makes maintenance of docstrings a bit easier, and reduces the number of lines of docstrings we need to scroll through before we hit the code. If we were to start expanding the docstrings, I would probably want to leverage another sprint to expand them all at once, so until we make that decision, for now it should be the minimal list above.

Reply:
Thanks, @ericmjl.  I updated the doc string using the example from the biology submodule.  I did not create an example for this PR.  I figure in the future I will create another PR (with an associated issue) that contains a single example notebook that covers both of the finance submodule's methods: convert_currency() and inflate_currency().

I think this PR should be ready now.

Reply:
Awesome stuff, I just reviewed the changes, and they look great to me!